### DISCUSSION TITLE
token User Can't reach login page.

### DISCUSSION THREAD

**Discussion by israaaa270** [ory/hydra#2346] [Q&A] [answered]
when I run token user  I can't reach the login page.
http://127.0.0.1:3000/login?login_challenge=c5cffd7dd9a94e8190da56da627c9658

information:
hydra version:` oryd/hydra:v1.9.0`
consent version: `v1.9.2`

-I run from `Terminal binary hydra` , NO DOCKER.
-create client:
`hydra clients create --endpoint http://127.0.0.1:4445 --id auth-code-client --secret secret --grant-types authorization_code,refresh_token  --response-types code,id_token  --scope openid,offline --callbacks http://127.0.0.1:5555/callback`
-token User: 
` hydra token user --client-id auth-code-client --client-secret secret --endpoint http://127.0.0.1:4444/  --port 5555  --scope openid,offline`
    open this link : `http://127.0.0.1:5555/`
-click on Authorize Application.
- redirect into 
`http://127.0.0.1:4444/oauth2/auth?audience=&client_id=auth-code-client&max_age=0&nonce=wqyunguosinlnqlhbpsbmruz&prompt=&redirect_uri=http%3A%2F%2F127.0.0.1%3A5555%2Fcallback&response_type=code&scope=openid+offline&state=bvrvjvrzddnpdneqsjhqijzz `     => status: 500

open this page : `http://127.0.0.1:3000/login?login_challenge=c5cffd7dd9a94e8190da56da627c9658`

with error : 
`{ "message": "connect ECONNREFUSED 127.0.0.1:4445", "name": "Error", "stack": "Error: connect ECONNREFUSED 127.0.0.1:4445\n at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:16)", "config": { "url": "https://127.0.0.1:4445/oauth2/auth/requests/login?login_challenge=c5cffd7dd9a94e8190da56da627c9658", "method": "get", "headers": { "Accept": "application/json, text/plain, */*", "User-Agent": "axios/0.19.2" }, "transformRequest": [ null ], "transformResponse": [ null ], "timeout": 0, "xsrfCookieName": "XSRF-TOKEN", "xsrfHeaderName": "X-XSRF-TOKEN", "maxContentLength": -1 }, "code": "ECONNREFUSED" } `

`in consent :`
Error: connect ECONNREFUSED 127.0.0.1:4445
at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:16)
GET /login?login_challenge=539d1442041d42f59fb7d147801a4928 500 360.029 ms - 1036
Error: Not Found
at /usr/src/app/lib/app.js:32:10
at Layer.handle [as handle_request] (/usr/src/app/node_modules/express/lib/router/layer.js:95:5)
at trim_prefix (/usr/src/app/node_modules/express/lib/router/index.js:317:13)
at /usr/src/app/node_modules/express/lib/router/index.js:284:7
at Function.process_params (/usr/src/app/node_modules/express/lib/router/index.js:335:12)
at next (/usr/src/app/node_modules/express/lib/router/index.js:275:10)
at /usr/src/app/node_modules/express/lib/router/index.js:635:15
at next (/usr/src/app/node_modules/express/lib/router/index.js:260:14)
at Function.handle (/usr/src/app/node_modules/express/lib/router/index.js:174:3)
at router (/usr/src/app/node_modules/express/lib/router/index.js:47:12)
GET /favicon.ico 500 29.527 ms - 80

I guess this error because
 `  "config": { "url": "https://127.0.0.1:4445/oauth2/auth/requests/login?login_challenge=c5cffd7dd9a94e8190da56da627c9658",`

**Comment by vinckr** [comment, marked as answer]
What does this part mean? 
>I guess this error because
"config": { "url": "https://127.0.0.1:4445/oauth2/auth/requests/login?login_challenge=c5cffd7dd9a94e8190da56da627c9658",

Can you post your hydra.yaml?

**Reply by israaaa270** [reply]
this error appear , this is the part mean .....
{ "message": "connect ECONNREFUSED 127.0.0.1:4445", "name": "Error", "stack": "Error: connect ECONNREFUSED 127.0.0.1:4445\n at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:16)",

`"config": { "url": "https://127.0.0.1:4445/oauth2/auth/requests/login?login_challenge=c5cffd7dd9a94e8190da56da627c9658", `

"method": "get", "headers": { "Accept": "application/json, text/plain, */*", "User-Agent": "axios/0.19.2" }, "transformRequest": [ null ], "transformResponse": [ null ], "timeout": 0, "xsrfCookieName": "XSRF-TOKEN", "xsrfHeaderName": "X-XSRF-TOKEN", "maxContentLength": -1 }, "code": "ECONNREFUSED" }

can't connect to "https://127.0.0.1:4445/oauth2/auth/requests/login?login_challenge=c5cffd7dd9a94e8190da56da627c9658"

hydra.yaml :

```
serve:
  cookies:
    same_site_mode: Lax

urls:
  self:
    issuer: http://127.0.0.1:4444
  consent: http://127.0.0.1:3000/consent
  login: http://127.0.0.1:3000/login
  logout: http://127.0.0.1:3000/logout

secrets:
  system:
    - youReallyNeedToChangeThis

oidc:
  subject_identifiers:
    supported_types:
      - pairwise
      - public
    pairwise:
      salt: youReallyNeedToChangeThis
```

**Reply by vinckr** [reply]
Can you check if your `HYDRA_ADMIN_URL` is https://127.0.0.1:4445 ? 

If thats not it:
`connect ECONNREFUSED 127.0.0.1:4445` seems to be a node.js error. 
Can you double check that you dont mix up localhost and 127.0.0.1 somewhere in your node setup? 
Other than that, maybe there is a problem with your port 4445/3000 that I am not seeing

**Reply by israaaa270** [reply]
Yes, I check when I run consent container in docker I set HYDRA_ADMIN_URL is https://127.0.0.1:4445 
 `
docker run -d 
  --name hydra-consent 
  -p 3000:3000 
  --network hydraguide 
  -e HYDRA_ADMIN_URL=https://127.0.0.1:4445 
  -e NODE_TLS_REJECT_UNAUTHORIZED=0 
  oryd/hydra-login-consent-node:v1.9.2
`

**Reply by vinckr** [reply]
Hmm. 
Is ` NODE_TLS_REJECT_UNAUTHORIZED=0` working correctly? 
I am not sure what the problem is. 
Would you be open to share a repo with your setup so I can clone and reproduce?

**Reply by vinckr** [reply]
Hey @israaaa270 are you still stuck on this?

**Reply by israaaa270** [reply]
No , thank you @vinckr  I passed it.

**Reply by rm116** [reply]
@israaaa270  How did you solve it ?
